gotest_plain_me/BBQ_test.go

```diff
@@ -0,0 +1,99 @@
+package gotestplainme
+
+import (
+	mock_beeffarm "github.com/artiomgiza/test_me/pkgs/beef-farm/mock"
+	mock_chickenfarm "github.com/artiomgiza/test_me/pkgs/chicken-farm/mock"
+	mock_coalfarm "github.com/artiomgiza/test_me/pkgs/coal-farm/mock"
+	mock_mangalstore "github.com/artiomgiza/test_me/pkgs/mangal-store/mock"
+	"github.com/golang/mock/gomock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+var (
+	mockBeefFarm    *mock_beeffarm.MockProvider
+	mockChickenFarm *mock_chickenfarm.MockProvider
+	mockMangalStore *mock_mangalstore.MockProvider
+	mockCoatFarm    *mock_coalfarm.MockProvider
+
+	subject coolPriceCalculator
+)
+
+func initMocks(t *testing.T) (finish func()) {
+	mockCtrl := gomock.NewController(t)
+	mockBeefFarm = mock_beeffarm.NewMockProvider(mockCtrl)
+	mockChickenFarm = mock_chickenfarm.NewMockProvider(mockCtrl)
+	mockMangalStore = mock_mangalstore.NewMockProvider(mockCtrl)
+	mockCoatFarm = mock_coalfarm.NewMockProvider(mockCtrl)
+
+	subject = coolPriceCalculator{
+		beefFarm:    mockBeefFarm,
+		chickenFarm: mockChickenFarm,
+		mangalStore: mockMangalStore,
+		coalFarm:    mockCoatFarm,
+	}
+	return func() {
+		mockCtrl.Finish()
+	}
+}
+
+func Test_coolPriceCalculator_CalculatePrice_Gotest(t *testing.T) {
+	t.Run("when people counter is exceeded max value", func(t *testing.T) {
```

Comment:
Lack of `defer initMocks(t)()` leads to `panic: runtime error: invalid memory address or nil pointer dereference` and aborting the following tests.